### ISSUE TITLE
Remind me Today -> Starts tomorrow

### ISSUE THREAD

**Issue by Fmyschkin** [scottleedavis/mattermost-plugin-remind#176]
#### Summary
/remind me test today -> schedules reminder for tomorrow
/remind me test at 18:00 -> schedules reminder for tomorrow
/remind me test at 18:00 today -> schedules reminder for tomorrow

#### Steps to reproduce
1. /remind me test today
2. Reminder is set tomorrow 9am

1. /remind me test at 18:00 (Current Time 15:00)
2. Reminder is set tomorrow 6pm

1. /remind me test at 18:00 today (Current Time 15:11)
2. Reminder is set tomorrow 6pm

#### Expected behavior
Reminder should be set to today

#### Server configuration
**Operating system:**
Linux Ubuntu

**Mattermost version:**
5.18.1

**Remind version:**
0.4.4
![remindbot_bug](https://user-images.githubusercontent.com/16978128/74943650-08536f00-53f5-11ea-84c7-d7b72a57fc07.gif)

#### Additional context
Experimental timezone support is enabled.

**Comment by piotr-wegert** [comment]
Happens to us as well.

**Comment by YuKitAs** [comment]
I think the issue lies with the rounding
```go
t, pErr := time.ParseInLocation(time.Kitchen, strings.ToUpper(normalizedWhen), location)
now := time.Now().In(location).Round(time.Hour * time.Duration(24))
occurrence := t.AddDate(now.Year(), int(now.Month())-1, now.Day()-1)
```
If `now` is later than 12:00 in UTC, it will be rounded up the next day, and the calculation for the occurrence will be incorrect. For example:
```go
now: 2020-05-24 15:10:17.815540836 +0200 CEST
rounded now: 2020-05-25 02:00:00 +0200 CEST
```
In this case the `occurrence` should be `t.AddDate(now.Year(), int(now.Month())-1, now.Day()-2)` imho.

**Comment by lucaswerkmeister** [comment]
Since it hasn’t been mentioned here yet, a **workaround** is to explicitly mention today’s date:

> /remind me test on <var>16.11.2020</var> at 18:00

**Comment by scottleedavis** [comment]
Fixed occurence calculation #193
